pkg/apis/resource/types.go

```diff
@@ -128,21 +128,28 @@ type ResourceClaimStatus struct {
 // claim.status.reservedFor.
 const ResourceClaimReservedForMaxSize = 32
 
-// AllocationResult contains attributed of an allocated resource.
+// AllocationResult contains attributes of an allocated resource.
 type AllocationResult struct {
-	// ResourceHandle contains arbitrary data returned by the driver after a
-	// successful allocation. This is opaque for
-	// Kubernetes. Driver documentation may explain to users how to
-	// interpret this data if needed.
+	// Context contains any state associated with the allocation that should be
+	// maintained throughout the lifetime of the claim. It is stored as a set
+	// of ResourceHandles, each of which contains data that should be passed to
+	// a specific kubelet plugin. This data is returned by the driver after a
+	// successful allocation and is opaque to Kubernetes.  Driver documentation
+	// may explain to users how to interpret this data if needed.
 	//
-	// The maximum size of this field is 16KiB. This may get
-	// increased in the future, but not reduced.
+	// Setting this field is optional. If null (or empty), it is assumed this
+	// allocation will be processed by a single kubelet plugin with no
+	// ResourceHandle data attached. The name of the kubelet plugin invoked
+	// will match the DriverName set in the ResourceClaimStatus this
+	// AllocationResult is embedded in.
+	//
+	// +listType=atomic
```

Comment:
if I can have N ResourceHandles, I can grow very large very quickly.  What is the contractual upper-bound here?  4?  8?  It can't be unbounded.  I see it below - comment here?

Reply:
Updated